Meadowlark-js/packages/meadowlark-core/src/handler/FrontendFacade.ts

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+import R from 'ramda';
+import { writeErrorToLog } from '../Logger';
+import { authenticate } from '../middleware/AuthenticationMiddleware';
+import { MiddlewareChain } from '../middleware/MiddlewareChain';
+import { parsePath } from '../middleware/ParsePathMiddleware';
+import { parseBody } from '../middleware/ParseBodyMiddleware';
+import { validateResource } from '../middleware/ValidateResourceMiddleware';
+import { noBody } from '../middleware/NoBodyMiddleware';
+import { FrontendRequest } from './FrontendRequest';
+import { FrontendResponse } from './FrontendResponse';
+import * as Upsert from './Upsert';
+import * as Update from './Update';
+import * as Delete from './Delete';
+import * as Get from './Get';
+
+// Middleware stack for methods with a body in the request (POST, PUT)
+const methodWithBodyStack: (chain: MiddlewareChain) => Promise<MiddlewareChain> = R.pipe(
+  authenticate,
+  R.andThen(parsePath),
+  R.andThen(parseBody),
+  R.andThen(validateResource),
+);
+
+// Middleware stack for methods without a body in the request (GET, DELETE)
+const methodWithoutBodyStack: (chain: MiddlewareChain) => Promise<MiddlewareChain> = R.pipe(
+  authenticate,
+  R.andThen(parsePath),
+  R.andThen(noBody),
+  R.andThen(validateResource),
+);
+
+/**
+ * Entry point for all API GET requests, both "by id" and query
+ */
+export async function getResolver(frontendRequest: FrontendRequest): Promise<FrontendResponse> {
+  try {
+    const chain: MiddlewareChain = await methodWithoutBodyStack({ frontendRequest, frontendResponse: null });
```

Comment:
Middlewares take a MiddlewareChain, which just a frontend request and frontend response. Each middleware operates on the request, possibly enriching it, then passing it on. If a middleware want to end processing, it also passes on a response. If a middleware sees a response in the chain, it should do nothing but passthrough the chain.

Here, the stack is called with the request and no response. If there is a response from a middleware, it's returned. Otherwise, the getResolver handler is called with the now-enriched frontend request.